ChangeLog

```diff
@@ -10,6 +10,10 @@ Release date: TBA
   Put new features here and also in 'doc/whatsnew/2.14.rst'
 
 
+* Exclude ``unnecessary-dunder-call`` on uninstantiated classes and within dunder funcdefs.
+
+  Closes #6074
+
 * Exclude dunder calls on super() from raising ``unnecessary-dunder-call``.
 
   Closes #6074
```

Comment:
```suggestion
```
I think we can remove fixes we do before releasing from the changelog and add the "Closes #6074" after Closes #5936 line 56.